Multisig/Logic/Models/Transaction.swift

```diff
@@ -45,21 +50,37 @@ extension Transaction {
         safeTxHash = multiSigTxInfo.safeTxHash
     }
 
-    var safeEncodedTxData: Data {
-        [
-            Safe.DefaultEIP712SafeAppTxTypeHash,
-            to.data32,
-            value.data32,
-            EthHasher.hash(data.data),
-            operation.data32,
-            safeTxGas.data32,
-            baseGas.data32,
-            gasPrice.data32,
-            gasToken.data32,
-            refundReceiver.data32,
-            nonce.data32
-        ]
-        .reduce(Data()) { $0 + $1 }
+    init?(wcRequest: WCSendTransactionRequest) {
+        dispatchPrecondition(condition: .notOnQueue(.main))
+
+        safe = wcRequest.from
+
+        var _nonce: UInt256String
+        if let latestTx = try? App.shared.safeTransactionService.latestTransaction(for: wcRequest.from) {
```

Comment:
please add comment why we're checking latest transaction

Reply:
ok